### PR TITLE
Allow previous versions of ciso8601 and requests, Unit test for null handling

### PR DESCRIPTION
**Pull request by abhi19gupta**
- Loose dependency requirements for ciso8601 and requests.
- Adding Unit test for null handling

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,10 +10,10 @@ homepage = "https://github.com/python-pinot-dbapi/pinot-dbapi"
 
 [tool.poetry.dependencies]
 python = ">=3.7,<4"
-ciso8601 = "^2.2.0"
+ciso8601 = "^2.1.3"
```
Comment (walterddr) [inline pyproject.toml:13]: question: this means exactly 2.1.3 or higher than or equal to 2.1.3? how does it work?

**Review by walterddr — APPROVED** [review]
lgtm thank you for fixing this. could you elaborate more on the PR description so we can reference back next time we upgrade versions. 

also CC @diogobaeder for the poetry version changes.

**Review by xiangfu0 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (xiangfu0) [reply pyproject.toml:13]: Ref: https://python-poetry.org/docs/dependency-specification/

Comment (diogobaeder) [reply pyproject.toml:13]: That means any version >= 2.1.3, but lower than 3.0.0.

**Comment by diogobaeder** [comment]
@walterddr I just replied above, thanks for the ping :-)

File: pyproject.toml (hunk 1 above)
Comment (walterddr) [reply pyproject.toml:13]: thanks for the answers, i think my original question is more on install time / env.
1. will it automatically pull 2.1.3, or the newest at the time of install? 
2. if a pinot-dbapi with 2.2.0 installation exist and we upgrade, will we reuse the 2.2.0 or the answer from question 1?
3. if another dependency pins the version to 2.2.0 within the same python venv, does it attempt to install 2.1.3? does it keep 2 versions and symlink them differently?

sorry for the confusions :-/

Comment (diogobaeder) [reply pyproject.toml:13]: No worries at all, man, those questions are an opportunity for everyone to learn more about that tool - including myself! :-)

So, the dependencies in the `pyproject.toml` file are not pinned dependencies, they just define version ranges for the dependencies. The pinned dependencies, however, are defined in the `poetry.lock` file instead. And it seems like this PR missed updating the lock file (with command `$ poetry lock`). So, for example, you might end up seeing, in the lock file, something like version `2.3.4`, but `^2.1.3` in `pyproject.toml` (because it only defines a loose dependency, not a pinned one).
